service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -298,6 +298,8 @@ public void open() throws IOException
             throw new IOException("AMQP sender is not initialized");
         }
 
+        this.executor = Executors.newFixedThreadPool(10);
```

Comment:
We can't simply re-open a previously used executor service since they are single use and can't be re-used after they are closed.

Reply:
When would we need to reopen/reinitialize this executor service? Do service clients follow open-close-open flow?

Reply:
We do allow for users to re-open a closed service client, yes. As long as they call open, the executor service is re-initialized for them. Users can't do open -> close -> sendAsync, for example. But they can do open -> close -> open -> sendAsync